subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/StronglyTypedConfigurationAttributesResolveIntegrationTest.groovy

```diff
@@ -400,159 +392,4 @@ class StronglyTypedConfigurationAttributesResolveIntegrationTest extends Abstrac
         notExecuted ':b:fooJar', ':b:barJar', ':b:bar2Jar'
     }
 
-    def "can select configuration thanks to producer schema disambiguation"() {
-        given:
-        file('settings.gradle') << "include 'a', 'b'"
-        buildFile << """
-            $typeDefs
-
-            project(':b') {
-               configurationAttributesSchema {
-                  attribute(buildType)
-                  attribute(flavor) {
-                       disambiguationRules.add { details ->
-                            details.closestMatch(details.candidateValues.sort { it.get() }.first())
```

Comment:
We can't because the disambiguation rule now has `T` as consumer value. And for this use case, `T` doesn't exist. `null` is not allowed, and if it was, I would consider this a design flaw. In other words, either we want to use `AttributeValue`, or not. But not a half baked solution.